src/index.js

```diff
@@ -1,65 +1,16 @@
-import firebase from 'firebase/compat/app';
-import * as firebaseui from 'firebaseui'
-import 'firebaseui/dist/firebaseui.css'
+import Vue from "vue";
 
-// Your web app's Firebase configuration
-const firebaseConfig = {
-  apiKey: "AIzaSyB3_ktYCDAvZR8ViEgmkKT8CDMPm1Q53Zo",
-  authDomain: "safetydungeon.firebaseapp.com",
-  projectId: "safetydungeon",
-  storageBucket: "safetydungeon.appspot.com",
-  messagingSenderId: "156587932337",
-  appId: "1:156587932337:web:116ae943f4f2ff92790090"
-};
+import "firebaseui/dist/firebaseui.css";
 
-// Initialize Firebase
-const app = firebase.initializeApp(firebaseConfig);
-//const db = getFirestore(app);
+import App from "./components/App";
 
-// Initialize the FirebaseUI Widget using Firebase.
-const ui = new firebaseui.auth.AuthUI(firebase.auth());
+import router from "./router";
+import store from "./store";
 
-// Add the email provider ID to the list of FirebaseUI signInOptions
-// ui.start('#firebaseui-auth-container', {
-//   signInOptions: [{
-//     provider: firebase.auth.EmailAuthProvider.PROVIDER_ID,
-//     requireDisplayName: false
-//   }]
-// });
+Vue.config.productionTip = false;
 
-// Specify FirebaseUI config
-const uiConfig = {
-  callbacks: {
-    signInSuccessWithAuthResult: function (authResult, redirectUrl) {
-      console.log("great success!", authResult)
-      // User successfully signed in.
-      // Return type determines whether we continue the redirect automatically
-      // or whether we leave that to developer to handle.
-      return true;
-    },
-    uiShown: function () {
-      // The widget is rendered.
-      // Hide the loader.
-      document.getElementById('loader').style.display = 'none';
-    }
-  },
-  // Will use popup for IDP Providers sign-in flow instead of the default, redirect.
-  signInFlow: 'popup',
-  signInSuccessUrl: '/',
-  signInOptions: [
-    // Leave the lines as is for the providers you want to offer your users.
-    firebase.auth.EmailAuthProvider.PROVIDER_ID,
-    // firebase.auth.GoogleAuthProvider.PROVIDER_ID,
-    // firebase.auth.FacebookAuthProvider.PROVIDER_ID,
-    // firebase.auth.TwitterAuthProvider.PROVIDER_ID,
-    // firebase.auth.GithubAuthProvider.PROVIDER_ID,
-    // firebase.auth.PhoneAuthProvider.PROVIDER_ID
-  ],
-  // Terms of service url.
-  tosUrl: '<your-tos-url>',
-  // Privacy policy url.
-  privacyPolicyUrl: '<your-privacy-policy-url>'
-};
-
-// The start method will wait until the DOM is loaded.
-ui.start('#firebaseui-auth-container', uiConfig);
\ No newline at end of file
+new Vue({
```

Comment:
These five lines are how we create a new instance of Vue. It gets the whole thing started

But I don't understand anything else about it. Why is it render? What does the function do? Maybe the .$mount has something to do with making all this happen early in page load and it has something to do with the #app DOM element in index.html

Reply:
> But I don't understand anything else about it.

Neither do I really. And that's ok. It's ok to not understand every line of a codebase and only learn about it when you need to.